### PR TITLE
docs: sync and translation of learn/forms

### PR DESCRIPTION
**Pull request by jasonren0403**
New translation: 
1. html5_input_types
2. other_form_controls

Sync translation:
1. HTML_form_in_legacy_browsers
2. How_to_build_custom_form_controls

### CODE REVIEW

File: files/zh-cn/learn/forms/how_to_build_custom_form_controls/example_1/index.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
 ---
-title: Example 1
+title: 示例 1
 slug: Learn/Forms/How_to_build_custom_form_controls/Example_1
 original_slug: Learn/HTML/Forms/How_to_build_custom_form_widgets/Example_1
 ---
-这是第一个[如果构建自定义表单小部件](/en-US/docs/Web/Guide/HTML/Forms/How_to_build_custom_form_widgets)的代码解释事例。
+这是第一个[如何构建自定义表单小部件](/zh-CN/docs/Web/Guide/HTML/Forms/How_to_build_custom_form_widgets)的代码解释事例。
```
Comment (jasonren0403) [inline files/zh-cn/learn/forms/how_to_build_custom_form_controls/example_1/index.md:6]:
```suggestion
这是第一个[如何构建自定义表单小部件](/zh-CN/docs/Learn/Forms/How_to_build_custom_form_widgets)的代码解释事例。
```

File: files/zh-cn/learn/forms/how_to_build_custom_form_controls/example_2/index.md [hunk 2]
```diff
@@ -1,9 +1,9 @@
 ---
-title: Example 2
+title: 示例 2
 slug: Learn/Forms/How_to_build_custom_form_controls/Example_2
 original_slug: Learn/HTML/Forms/How_to_build_custom_form_widgets/Example_2
 ---
-这是解释 [如何构建自定义表单小部件](/en-US/docs/HTML/Forms/How_to_build_custom_form_widgets)的第二个示例。
+这是解释[如何构建自定义表单小部件](/zh-CN/docs/HTML/Forms/How_to_build_custom_form_widgets)的第二个示例。
```
Comment (jasonren0403) [inline files/zh-cn/learn/forms/how_to_build_custom_form_controls/example_2/index.md:6]:
```suggestion
这是解释[如何构建自定义表单小部件](/zh-CN/docs/Learn/Forms/How_to_build_custom_form_widgets)的第二个示例。
```

File: files/zh-cn/learn/forms/how_to_build_custom_form_controls/example_3/index.md [hunk 3]
```diff
@@ -1,9 +1,9 @@
 ---
-title: Example 3
+title: 示例 3
 slug: Learn/Forms/How_to_build_custom_form_controls/Example_3
 original_slug: Learn/HTML/Forms/How_to_build_custom_form_widgets/Example_3
 ---
-这是解释 [如何构建自定义表单小部件](/en-US/docs/HTML/Forms/How_to_build_custom_form_widgets) 的第三个示例。
+这是解释[如何构建自定义表单小部件](/zh-CN/docs/HTML/Forms/How_to_build_custom_form_widgets)的第三个示例。
```
Comment (jasonren0403) [inline files/zh-cn/learn/forms/how_to_build_custom_form_controls/example_3/index.md:6]:
```suggestion
这是解释[如何构建自定义表单小部件](/zh-CN/docs/Learn/Forms/How_to_build_custom_form_widgets)的第三个示例。
```

File: files/zh-cn/learn/forms/how_to_build_custom_form_controls/example_4/index.md [hunk 4]
```diff
@@ -1,9 +1,9 @@
 ---
-title: Example 4
+title: 示例 4
 slug: Learn/Forms/How_to_build_custom_form_controls/Example_4
 original_slug: Learn/HTML/Forms/How_to_build_custom_form_widgets/Example_4
 ---
-这是解释 [如何构建自定义表单小部件](/en-US/docs/Learn/HTML/Forms/How_to_build_custom_form_widgets) 的第四个示例。
+这是解释[如何构建自定义表单小部件](/zh-CN/docs/Learn/HTML/Forms/How_to_build_custom_form_widgets)的第四个示例。
```
Comment (jasonren0403) [inline files/zh-cn/learn/forms/how_to_build_custom_form_controls/example_4/index.md:6]:
```suggestion
这是解释[如何构建自定义表单小部件](/zh-CN/docs/Learn/Forms/How_to_build_custom_form_widgets)的第四个示例。
```

File: files/zh-cn/learn/forms/how_to_build_custom_form_controls/example_3/index.md [hunk 5]
```diff
@@ -167,80 +167,86 @@ original_slug: Learn/HTML/Forms/How_to_build_custom_form_widgets/Example_3
 // ------- //
 
 NodeList.prototype.forEach = function (callback) {
-  Array.prototype.forEach.call(this, callback);
+    Array.prototype.forEach.call(this, callback);
```
Comment (yin1999) [inline files/zh-cn/learn/forms/how_to_build_custom_form_controls/example_3/index.md:170]: It seems that this content is not synced with english document?

And we just use two spaces as indentation

<https://developer.mozilla.org/en-US/docs/Learn/Forms/How_to_build_custom_form_controls/Example_3>

File: files/zh-cn/learn/forms/html5_input_types/index.md [hunk 6]
```diff
@@ -0,0 +1,288 @@
...
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+> **备注：** 本篇文章中讨论的大多数特性都受到了广泛支持，如果有任何例外将会在文章中说明，如果你需要更多浏览器支持的细节，你应该查看我们的 [HTML 表单元素参考](/zh-CN/docs/Web/HTML/Element#forms)，特别是深入的 [\<input> 类型](/zh-CN/docs/Web/HTML/Element/Input) 参考。
+
+由于 HTML 表单控件的外观可能与设计者的规格有很大的不同，web 开发者有时会建立自己的自定义表单控件。我们在一个高级教程中介绍了这一点：[如何搭建自定义表单控件](/zh-CN/docs/Learn/Forms/How_to_build_custom_form_controls)。
+
+## E-mail 地址域
```
Comment (yin1999) [inline files/zh-cn/learn/forms/html5_input_types/index.md:33]: should we use "字段"?

```suggestion
## E-mail 地址字段
```

File: files/zh-cn/learn/forms/other_form_controls/index.md [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+    <option>Blueberry</option>
+    <option>Lemon</option>
+    <option>Lychee</option>
+    <option>Peach</option>
+    <option>Pear</option>
+  </select>
+</datalist>
+```
+
+{{EmbedLiveSample("Datalist_support_and_fallbacks", 120, 120)}}
```
Comment (yin1999) [inline files/zh-cn/learn/forms/other_form_controls/index.md:218]: the firest param should match with heads:

```suggestion
{{EmbedLiveSample("Datalist 支持和回退", 120, 120)}}
```

File: files/zh-cn/learn/forms/other_form_controls/index.md [hunk 8]
```diff
@@ -0,0 +1,325 @@
...
+
+#### 进度条
+
+一个进度条表示一个随时间变化的竖直，最高可以达到 {{htmlattrxref("max","progress")}} 属性所指定的最大值。这样的条是使用 {{ HTMLElement("progress")}} 元素创建出来的。
+
+```html
+<progress max="100" value="75">75/100</progress>
+```
+
+{{EmbedLiveSample("Progress", 120, 120)}}
```
Comment (yin1999) [inline files/zh-cn/learn/forms/other_form_controls/index.md:256]: same

File: files/zh-cn/learn/forms/how_to_build_custom_form_controls/example_1/index.md [hunk 9]
```diff
@@ -1,9 +1,9 @@
 ---
-title: Example 1
+title: 示例 1
 slug: Learn/Forms/How_to_build_custom_form_controls/Example_1
 original_slug: Learn/HTML/Forms/How_to_build_custom_form_widgets/Example_1
 ---
-这是第一个[如果构建自定义表单小部件](/en-US/docs/Web/Guide/HTML/Forms/How_to_build_custom_form_widgets)的代码解释事例。
+这是第一个[如何构建自定义表单小部件](/zh-CN/docs/Learn/Forms/How_to_build_custom_form_widgets)的代码解释事例。
```
Comment (jasonren0403) [inline files/zh-cn/learn/forms/how_to_build_custom_form_controls/example_1/index.md:6]:
```suggestion
这是第一个[如何构建自定义表单小部件](/zh-CN/docs/Learn/Forms/How_to_build_custom_form_controls)的代码解释事例。
```

**Review by yin1999 — CHANGES_REQUESTED** [review]
some general suggestions

And could you split the changes into multiple PRs?

File: files/zh-cn/learn/forms/how_to_build_custom_form_controls/example_2/index.md [hunk 10]
```diff
@@ -1,9 +1,9 @@
 ---
-title: Example 2
+title: 示例 2
 slug: Learn/Forms/How_to_build_custom_form_controls/Example_2
 original_slug: Learn/HTML/Forms/How_to_build_custom_form_widgets/Example_2
 ---
-这是解释 [如何构建自定义表单小部件](/en-US/docs/HTML/Forms/How_to_build_custom_form_widgets)的第二个示例。
+这是解释[如何构建自定义表单小部件](/zh-CN/docs/Learn/Forms/How_to_build_custom_form_widgets)的第二个示例。
```
Comment (jasonren0403) [inline files/zh-cn/learn/forms/how_to_build_custom_form_controls/example_2/index.md:6]:
```suggestion
这是解释[如何构建自定义表单小部件](/zh-CN/docs/Learn/Forms/How_to_build_custom_form_controls)的第二个示例。
```

File: files/zh-cn/learn/forms/how_to_build_custom_form_controls/example_3/index.md [hunk 11]
```diff
@@ -1,9 +1,9 @@
 ---
-title: Example 3
+title: 示例 3
 slug: Learn/Forms/How_to_build_custom_form_controls/Example_3
 original_slug: Learn/HTML/Forms/How_to_build_custom_form_widgets/Example_3
 ---
-这是解释 [如何构建自定义表单小部件](/en-US/docs/HTML/Forms/How_to_build_custom_form_widgets) 的第三个示例。
+这是解释[如何构建自定义表单小部件](/zh-CN/docs/Learn/Forms/How_to_build_custom_form_widgets)的第三个示例。
```
Comment (jasonren0403) [inline files/zh-cn/learn/forms/how_to_build_custom_form_controls/example_3/index.md:6]:
```suggestion
这是解释[如何构建自定义表单小部件](/zh-CN/docs/Learn/Forms/How_to_build_custom_form_controls)的第三个示例。
```

File: files/zh-cn/learn/forms/how_to_build_custom_form_controls/example_4/index.md [hunk 12]
```diff
@@ -1,9 +1,9 @@
 ---
-title: Example 4
+title: 示例 4
 slug: Learn/Forms/How_to_build_custom_form_controls/Example_4
 original_slug: Learn/HTML/Forms/How_to_build_custom_form_widgets/Example_4
 ---
-这是解释 [如何构建自定义表单小部件](/en-US/docs/Learn/HTML/Forms/How_to_build_custom_form_widgets) 的第四个示例。
+这是解释[如何构建自定义表单小部件](/zh-CN/docs/Learn/Forms/How_to_build_custom_form_widgets)的第四个示例。
```
Comment (jasonren0403) [inline files/zh-cn/learn/forms/how_to_build_custom_form_controls/example_4/index.md:6]:
```suggestion
这是解释[如何构建自定义表单小部件](/zh-CN/docs/Learn/Forms/How_to_build_custom_form_controls)的第四个示例。
```

**Comment by jasonren0403** [comment]
Close to split it into multiple PRs

**Pull request closed without merging**